packages/site/cms-site/src/blocks/helpers/ErrorBoundary.tsx

```diff
@@ -0,0 +1,35 @@
+import React, { ErrorInfo, PropsWithChildren } from "react";
+
+type Props = {
+    fallback: React.ReactNode;
+    blockType: string;
+    reportError: (error: Error, errorInfo: ErrorInfo) => void;
+};
+
+type State = {
+    hasError: boolean;
+};
+
+export class ErrorBoundary extends React.Component<PropsWithChildren<Props>, State> {
```

Comment:
Have you considered using this package instead? https://www.npmjs.com/package/react-error-boundary

Reply:
No, I implemented it according to the [react docs](https://react.dev/reference/react/Component#catching-rendering-errors-with-an-error-boundary).

In my opinion it is straightforward enough for this case, but if we use the error boundary in more places in the future it could make sense. So if you want I'll use it.

Reply:
Let's leave it like this for now.